features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/ClientFactory.java

```diff
@@ -0,0 +1,144 @@
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2017-2017 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2017 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.flows.elastic;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import io.searchbox.client.AbstractJestClient;
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestClientFactory;
+import io.searchbox.client.config.HttpClientConfig;
+
+// TODO MVR merge with RestClientFactory in opennms-es-rest module/project
+
+// TODO MVR allow setting proxy
+// TODO MVR allow polling
+// TODO MVR allow setting password for each url, etc.
+// TODO MVR support for async
+public class ClientFactory {
+
+    private final HttpClientConfig.Builder clientConfigBuilder;
+
+    private JestClient client;
+
+    public ClientFactory(String elasticURL, String elasticUser, String elasticPassword) {
+        final List<String> urls = parseUrl(elasticURL);
+        final String user = elasticUser != null && !elasticUser.isEmpty() ? elasticUser : null;
+        final String password = elasticPassword != null && !elasticPassword.isEmpty() ? elasticPassword : null;
+
+        // Ensure urls is set
+        if (urls.isEmpty()) {
+            throw new IllegalArgumentException("No urls have been provided");
+        }
+
+        // TODO MVR Fix the following hack:
+        // This is a hack, due to our osgi-jetty-bridge, because the annotation at the NetflowDocument
+        // living in the api module (on the jetty side) are not accessible directly (only as proxy) by the Gson lirary
+        // living in the osgi container (osgi-jest-complete module) and therefore @SerializedName annotations won't work.
```

Comment:
If the `@SerializedName` annotations don't work, then why do we need the Gson dependency in the flow api module?

Reply:
The idea was to get gson to work, in order to define custom mappings. If that is not going to work in the end, we do not need this dependency, true.